[TFECO-12319] API DOCS For Registry Tagging and Visibility (#2682)
hashicorp/web-unified-docs: content/terraform-docs-common/docs/cloud-docs/api-docs/private-registry/providers/index.mdx

            }
          }
        ]
      }
    }
  }
}
```

### Sample Request

[](#sample-request)

```bash
curl \
  --header "Authorization: Bearer $TOKEN" \
  --header "Content-Type: application/vnd.api+json" \
  --request PATCH \
  --data @payload.json \
  https://app.terraform.io/api/v2/organizations/hashicorp/registry-providers/public/hashicorp/aws
```

### Sample Response

[](#sample-response)

```json
{
  "data": {
    "id": "prov-kdWEMRMEW20M",
    "type": "registry-providers",
    "attributes": {
      "name": "aws",
      "namespace": "hashicorp",
      "created-at": "2026-02-24T02:43:14.857Z",
      "updated-at": "2026-02-24T02:43:14.857Z",
      "registry-name": "private",
      "permissions": {
        "can-delete": true
      }
    },
    "relationships": {
      "organization": {
        "data": {
          "id": "hashicorp",
          "type": "organizations"
        }
      },
      "tag-bindings": {
        "links": {
          "related": "/api/v2/registry-providers/prov-kYepEWv1vNWdKAsM/tag-bindings"
        }
      }
    },
    "links": {
      "self": "/api/v2/organizations/hashicorp/registry-providers/private/hashicorp/test-provider"
    }
  }
}
```

> 

Reviewer: Add: `Add : <!-- BEGIN: TFC:only -->`

          {
            "key": "environment",
            "value": "testing"
          }
        ],
        "status": "",
        "name": "random",
        "namespace": "hashicorp"
      }
    }
  ],
  "links": {
    "self": "https://app.terraform.io/api/v2/organizations/RAUC/registry-providers/public/hashicorp/random/versions?filter%5Btagged%5D%5B0%5D%5Bkey%5D=environment&filter%5Btagged%5D%5B0%5D%5Bvalue%5D=testing&page%5Bnumber%5D=1&page%5Bsize%5D=20",
    "first": "https://app.terraform.io/api/v2/organizations/RAUC/registry-providers/public/hashicorp/random/versions?filter%5Btagged%5D%5B0%5D%5Bkey%5D=environment&filter%5Btagged%5D%5B0%5D%5Bvalue%5D=testing&page%5Bnumber%5D=1&page%5Bsize%5D=20",
    "prev": null,
    "next": null,
    "last": "https://app.terraform.io/api/v2/organizations/RAUC/registry-providers/public/hashicorp/random/versions?filter%5Btagged%5D%5B0%5D%5Bkey%5D=environment&filter%5Btagged%5D%5B0%5D%5Bvalue%5D=testing&page%5Bnumber%5D=1&page%5Bsize%5D=20"
  },
  "meta": {
    "pagination": {
      "current-page": 1,
      "page-size": 20,
      "prev-page": null,
      "next-page": null,
      "total-pages": 1,
      "total-count": 1
    }
  }
}

> ```

Reviewer: Add : `<!-- END: TFC:only -->`

| ------- | ------------------------- | -------------------------- |
| [200][] | [JSON API document][]     | The request was successful |
| [401][] | [JSON API error object][] | User unauthorized          |
| [404][] | [JSON API error object][] | Provider not found         |

### Sample Request

[](#sample-request-2)

```bash
curl \
  --request GET \
  --header "Authorization: Bearer $TOKEN" \
  https://app.terraform.io/api/v2/registry-providers/prov-fZn7uHu99ZCpAKZJ/tag-bindings
```

### Sample Response

[](#sample-response-2)

```json
{
	"data": [
		{
			"id": "tb-b1tr4235pc7eYCAV6a",
			"type": "tag-bindings",
			"attributes": {
				"key": "DEPT",
				"value": "ENG",
				"created-at": "2026-06-11T01:47:17.362Z"
			}
		},
		{
			"id": "tb-PqMJDFqrttmWYYSE",
			"type": "tag-bindings",
			"attributes": {
				"key": "env",
				"value": "staging",
				"created-at": "2026-03-03T00:12:39.529Z"
			}
		}
	],
	"links": {
		"target": "/api/v2/organizations/hashicorp/registry-providers/public/hashicorp/aws"
	}
}
```

## Add or Update Tags to a Registry Provider

[](#add-tags-to-a-registry-provider)

Adds or updates tags for a registry provider in your organization's registry.

> -> **Note:** Removing tags from a registry provider is made via PATCH requests, where the request body will contain only the tag key and values that are currently needed.

Reviewer 2: ```suggestion
Send a `PATCH` request to the following endpoint to adds or update tags for a registry provider in your organization's registry.

To remove tags from a registry module, omit the tags from the request body.
```

`PATCH /api/v2/organizations/:organization-name/registry-providers/:registry-name/:namespace/:provider-name`

| Parameter            | Description                                                                                                                                                       |
| -------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `:organization_name` | The name of the organization that owns the registry provider.                                                                                                    |
| `:registry_name`     | The name of the registry. For private providers, this should be `private` and for publicly curated providers, this should be `public`. Same for other artifacts. |

> | `:namespace`         | The namespace of the provider (typically your organization name for private providers).                                                                          |

Reviewer 2: ```suggestion
| `:namespace`         | The namespace of the provider. For private providers, the namespace is typically your organization name .                                                                          |
```

| `:provider-name`     | The name of the provider.                                                                                                                                        |

| Status  | Response                  | Reason                                                       |
| ------- | ------------------------- | ------------------------------------------------------------ |
| [200][] | [JSON API document][]     | Successfully added/updated the tag binding                   |
| [400][] | [JSON API error object][] | Request body cannot be parsed                                |
| [401][] | [JSON API error object][] | User not authorized                                          |
| [404][] | [JSON API error object][] | Registry provider not found                                  |

> | [422][] | [JSON API error object][] | Unprocessable content - Tag key or value have a wrong format |

Reviewer 2: ```suggestion
| [422][] | [JSON API error object][] | HCP Terraform can't process the content because the tag key or value format is incorrect.  |
```

### Request Body

[](#request-body)

> This PATCH endpoint requires a JSON object with the following properties as a request payload.

Reviewer 2: ```suggestion
To send a `PATCH` request to this endpoint, you must include the following properties as a request payload.
```

        ]
      }
    }
  }
}
```

### Sample Request

[](#sample-request)

```bash
curl \
  --header "Authorization: Bearer $TOKEN" \
  --header "Content-Type: application/vnd.api+json" \
  --request PATCH \
  --data @payload.json \
  https://app.terraform.io/api/v2/organizations/hashicorp/registry-providers/public/hashicorp/aws
```

### Sample Response

[](#sample-response)

```json
{
  "data": {
    "id": "prov-kdWEMRMEW20M",
    "type": "registry-providers",
    "attributes": {
      "name": "aws",
      "namespace": "hashicorp",
      "created-at": "2026-02-24T02:43:14.857Z",
      "updated-at": "2026-02-24T02:43:14.857Z",
      "registry-name": "private",
      "permissions": {
        "can-delete": true
      }
    },
    "relationships": {
      "organization": {
        "data": {
          "id": "hashicorp",
          "type": "organizations"
        }
      },
      "tag-bindings": {
        "links": {
          "related": "/api/v2/registry-providers/prov-kYepEWv1vNWdKAsM/tag-bindings"
        }
      }
    },
    "links": {
      "self": "/api/v2/organizations/hashicorp/registry-providers/private/hashicorp/test-provider"
    }
  }
}
```

<!-- BEGIN: TFC:only -->

> ## Versions List for a Registry Provider

Reviewer 2: ```suggestion
## List versions for a registry provider
```

[](#versions-list-for-a-registry-provider)

> This endpoint returns all the published versions for a registry provider with its tags' information.

Reviewer 2: ```suggestion
The following endpoint returns all published versions for a registry provider including tags attached to each version.
```

| `filter[tagged][INDEX][value]` | **Optional.** Filters versions by tag value.                                                                                         |
| `page[number]`                 | **Optional.** If omitted, the endpoint returns the first page.                                                                           |
| `page[size]`                   | **Optional.** If omitted, the endpoint returns 20 tag bindings per page.                                                                 |

When you provide multiple `filter[tagged]` conditions, the `INDEX` value determines how HCP Terraform combines them:

- Use the same index to combine a key and value with `AND`. For example, `filter[tagged][0][key]=environment&filter[tagged][0][value]=prod` matches tags where the key is `environment` and the value is `prod`.
- Use different indexes to combine conditions with `OR`. For example, `filter[tagged][0][key]=environment&filter[tagged][1][value]=prod` matches tags where the key is `environment` or the value is `prod`.

#### Sample request with query parameters

> The following request is very similar to the previous one, but it filters Provider Versions where the key is `environment` and the value is `testing`.

Reviewer 2: ```suggestion
The following request filters provider versions where the key is `environment` and the value is `testing`.
```
